### DISCUSSION TITLE
Hetzner Load Balancers Limits

### DISCUSSION THREAD

**Discussion by bgervan** [vitobotta/hetzner-k3s#248] [Q&A]
Hi everyone,

what are the options for load balancing if our cluster would need to handle more concurrent connection than the largest one by hetzner (40k), in case of websockets for example?

**Comment by vitobotta** [comment]
Have you checked if there is an annotation to select the balancer type? I won't be able to use a computer for some days so I can't check.

**Comment by derlinuxer** [comment]
You can set the balancer type via `load-balancer.hetzner.cloud/type` in you ingress controller config.

See doc: https://pkg.go.dev/github.com/hetznercloud/hcloud-cloud-controller-manager/internal/annotation#Name
```
// LBType specifies the type of the Load Balancer.
//
// Default: lb11.
LBType [Name](https://pkg.go.dev/github.com/hetznercloud/hcloud-cloud-controller-manager/internal/annotation#Name) = "load-balancer.hetzner.cloud/type"
```
The big one is lb31. If lb31 is not enough then you probably stuck on balancing with hetzner stuff.

**Reply by vitobotta** [reply]
Nice. How many connections can it handle?

**Reply by bgervan** [reply]
Name | Connections | Services | Targets | Certs | Price* |  
-- | -- | -- | -- | -- | -- | --
LB11 | 10000 | 5 | 25 | 10 | € 0.0088 /h | € 5.39 /mo
LB21 | 20000 | 15 | 75 | 25 | € 0.0253 /h | € 16.40 /mo
LB31 | 40000 | 30 | 150 | 50 | € 0.0495 /h | € 32.90 /mo

**Reply by derlinuxer** [reply]
Then you have to use a balancer from an more potential provider. But this isn't very comfortable by adding and removing nodes. I also read that hetzner FW seems to be limited by 80k connections. (isn't a reliable Statement)

**Reply by vitobotta** [reply]
Gotcha, thanks for sharing. Useful info 👍

**Reply by bgervan** [reply]
Yes I found this information here https://docs.hetzner.com/cloud/firewalls/overview/
"Up to 80000 active, concurrent connections per server (10000 new connections per second)"

**Comment by vitobotta** [comment]
@bgervan out of curiosity what kind of application are you working on that requires such a high number of connections?

**Reply by bgervan** [reply]
I am just exploring yet the capabilities. One of my app idea requires websocket connection to send notification and chat messages. But in case of hitting the limits, I could make something with https://pushpin.org/ as a workaround.
